### PR TITLE
feat(oidc): Akamai Hosted Login

### PR DESCRIPTION
**Pull request by VancheeZze**
## Changes
- New `@bodiless/oidc` package.

## Test Instructions
- Try to Log In and Log Out using the button on the `/oidc` page.
- You can use `bob/bob` or `alice/alice` for login/password

## Related Issues
Closes: https://github.com/johnsonandjohnson/Bodiless-JS/issues/1147

## Notes
- Currently uses https://demo.identityserver.io/ as an Identity Provider.

### CODE REVIEW

File: packages/bodiless-oidc/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "build:watch": "npm run build -- --watch",
+    "build:api-doc": "typedoc --out doc/api src",
+    "lint": "tslint ./src/**/*.ts?x",
+    "clean": "rimraf \"lib/*\" && rimraf tsconfig.tsbuildinfo && rimraf \"doc/api\""
+  },
+  "dependencies": {
+    "enzyme": "^3.9.0",
+    "oidc-client-ts": "^2.0.0-beta.2",
+    "axios": "^0.21.0",
+    "react-dom": "^17.0.2"
```
Comment (marcopagliarulo) [inline packages/bodiless-oidc/package.json:31]: I think react-dom should be a peerDepenency

Comment (VancheeZze) [reply packages/bodiless-oidc/package.json]: Addressed, thanks!

**Comment by TatsianaMiraniuk** [comment]
Hi @VancheeZze ,

please see **issue_1**:
User_1 that was previously logged in is logged in automatically after clicking Login button on the test /oidc/ page. /Account/Login page with Login Form doesn't appears.
The expected behavior could be discussed with the team. I would suppose there should be Log in form that allow to login with another user.
As of now, the workaround to get Login form is to remove  Cookies in the browser (Application -> Storage)

**Comment by TatsianaMiraniuk** [comment]
**issue_2** to discuss:

There are several params added in the URL after login that happens by click on Login.
Screen Shot 2021-11-03 at 5 09 34 PM

At the same time, there are no params added in the URL after login that happens via Login Popup.
Screen Shot 2021-11-03 at 5 09 06 PM

**Q**: can we keep the URL without the params since they are redundant?

**Comment by TatsianaMiraniuk** [comment]
> Hi @VancheeZze ,
> 
> please see **issue_1**: User_1 that was previously logged in is logged in automatically after clicking Login button on the test /oidc/ page. /Account/Login page with Login Form doesn't appears. The expected behavior could be discussed with the team. I would suppose there should be Log in form that allow to login with another user. _As of now, the workaround to get Login form after Logout is to stay on the page (NOT to reload the page) and remove Cookies in the browser (Application -> Storage). If the page reloads before removing Cookies manually, Cookies are deleted by the browser, but Login with the user_1 still happens automatically._
> 
> **Steps to reproduce:**
> 
> 1. open /oidc/ on Test Site
> 2. click Login -> https://demo.identityserver.io/Account/Login?....  opens
> 3. login as 'bob' -> redirection to /oidc/? happens
> 4. click Logout
> 5. click Login
> 
> **Actual Results:** the user is logged in automatically as 'bob'
> 
> **Expected Results:** (???)Login form opens


Fixed. Closing.

**Comment by TatsianaMiraniuk** [comment]
> **issue_2** to discuss:
> 
> There are several params added in the URL after login that happens by click on Login. Screen Shot 2021-11-03 at 5 09 34 PM
> 
> At the same time, there are no params added in the URL after login that happens via Login Popup. Screen Shot 2021-11-03 at 5 09 06 PM
> 
> **Q**: can we keep the URL without the params since they are redundant?

Fixed. Closing.

**Comment by TatsianaMiraniuk** [comment]
**issue_3:**

After Login a User is not redirected to the initial page.

**Steps to reproduce:**
1. open /oidc/ page
2. click Login
3. enter username/password
4. click Login

**Actual Results:**
A user lands on /oidc-redirect/

**Expected Results:**
A user lands on /oidc/ page where Login was initiated

**Comment by TatsianaMiraniuk** [comment]
**issue_4:**

After LogOut a User is not redirected to the initial page.

**Steps to reproduce:**
1. open /oidc/ page
2. click Login
3. enter username/password
4. click Login
5. open /oidc/ page
6. click Log Out

**Actual Results:**
A user lands on Identity Provider Log out page

**Expected Results:**
A user lands on /oidc/ page

**Comment by TatsianaMiraniuk** [comment]
**issue_5:**

There is an error in a console when a user click Cancel button on Identity Provider Login Form (in a case when that is done through the Popup)

**Steps to reproduce:**
1. open /oidc/ page
2. click Login Popup
3. click Cancel below Identity Provider Login Form
6. check a browser console on /oidc/ page

**Actual Results:**
there is a error: Uncaught (in promise) ErrorResponse: access_denied
Screen Shot 2021-11-04 at 5 08 52 PM


**Expected Results:**
no error in a console on /oidc/ page

**Comment by TatsianaMiraniuk** [comment]
issue_6:
Login/Logout button and info about the logged in user are only available on /oidc/ page. there is no possibility to confirm that a User is logged in/out on all other pages (e.g. it could be added in the Header)

**Comment by TatsianaMiraniuk** [comment]
**issue_7:**

**Steps to reproduce:**
1. open /oidc/ page
2. click Login
3. enter username/password as user_1
4. click Login
5. close a browser tab with /oidc/ page, but keep the browser open
6. open /oidc/ page

**Actual Results:**
Login button is shown. No user data visible

7. click Login

**Actual Results:**
A user lands on /oidc-redirect/. 

8. click /oidc/ link

**Actual Results:**
user data of user_1 is visible

**Expected Results:**
to be discussed

**Review by TatsianaMiraniuk — APPROVED** [review]

**Review by marcopagliarulo — APPROVED** [review]

**Pull request merged**